manta-pay/src/test/compat.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Manta Pay UTXO Binary Compatibility
+//! Checks if the current circuit implementation is compatible with precomputed parameters.
+
+use crate::{
+    config::{
+        MultiProvingContext, MultiVerifyingContext, NoteEncryptionScheme, Parameters,
+        ProvingContext, UtxoAccumulatorModel, UtxoCommitmentScheme, VerifyingContext,
+        VoidNumberCommitmentScheme,
+    },
+    payment::{assert_valid_proof, prove_mint, prove_private_transfer, prove_reclaim},
+};
+use anyhow::Result;
+use ark_std::rand::thread_rng;
+use manta_crypto::rand::Rand;
+use manta_util::codec::{Decode, IoReader};
+use std::{fs::File, path::Path};
+
+/// Loads parameters from the SDK, using `directory` as a temporary directory to store files.
+#[inline]
+fn load_parameters(
+    directory: &Path,
+) -> Result<(
+    MultiProvingContext,
+    MultiVerifyingContext,
+    Parameters,
+    UtxoAccumulatorModel,
+)> {
+    println!("Loading parameters...");
+    let mint_path = directory.join("mint.dat");
+    manta_parameters::pay::testnet::proving::Mint::download(&mint_path)?;
+    let private_transfer_path = directory.join("private-transfer.dat");
+    manta_parameters::pay::testnet::proving::PrivateTransfer::download(&private_transfer_path)?;
+    let reclaim_path = directory.join("reclaim.dat");
+    manta_parameters::pay::testnet::proving::Reclaim::download(&reclaim_path)?;
```

Comment:
Try to avoid `download` here. We have the files locally, so we should be able to read them directly.

Reply:
This `download` is only for lfs files in `ProvingVontext`. A developer does NOT have files locally unless he has intentionally run `git lfs pull` earlier.